internal/reconciler/metricpipeline/reconciler.go

```diff
@@ -238,3 +244,18 @@ func (r *Reconciler) getReplicaCountFromTelemetry(ctx context.Context) int32 {
 	}
 	return defaultReplicaCount
 }
+
+func getAllowedPorts(istioActive bool) []int32 {
```

Comment:
Agent and gateway ports should be different.

Reply:
done